src/main/ResolutionCache.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Cache for any Data Sources that rely on resolution. For example,
+ * this is used by coverage to keep track of the difference resolutions
+ * that have been fetched already.
+ *
+ * @flow
+ */
+'use strict';
+
+import _ from 'underscore';
+import Interval from './Interval';
+import ContigInterval from './ContigInterval';
+
+class ResolutionCache<T: Object> {
```

Comment:
I am not a huge fan of the terminology 'resolution' or 'binning' to refer to requests that explicitly specify resolution. @ryan-williams would this be useful for bringing in multiple resolution coverage in #430?